### ISSUE TITLE
predict function returning less number of predictions than expected.

### ISSUE THREAD

**Issue by run2** [benanne/Lasagne#95]
Hi
There is more information about this problem on my posts in Daniels blog.
Basically after fitting a net when I use the predict function, the number of rows coming back is lower than the number expected. So I have 3033 images on my test set. The predicted output has 3020 rows. This happens for different kinds of networks.

**Comment by benanne** [comment]
> May be at some point this issue can be re-opened to compute the predictions in batches for as many times as possible and then do it one by one for the remainder. Rather than asking the user to pad.

I suspect the code you're talking about is actually part of nolearn, because at the moment the only thing Lasagne does is generate Theano expressions. How those are used to do training / prediction is up to the user. The nolearn `NeuralNet` class takes care of some of that stuff if you choose to use it, So this issue should probably have been reported there instead.

We do plan to add some "training loop" code to Lasagne as well, at which point we'll definitely take into account this situation.

**Comment by dnouri** [comment]
Sorry for the confusion.  I guess I could've made a clearer distinction about what's Lasagne and what's nolearn.lasagne.

@run2: Maybe you're using this `forced_even` thing still?  That one was a bad idea (it's been removed in more recent versions); don't use that argument and just set the first dimension of `input_shape` to `None`.  I've updated the code in the blog post with this a while ago.  If this doesn't solve the issue, then open a bug in the nolearn tracker.
